app/controllers/shared/api.php

```diff
@@ -282,54 +282,85 @@
+            $user = $dbForProject->getDocument('users', $userId);
+            $audits->setUser($user);
         }
 
-        foreach ($events->getParams() as $key => $value) {
-            $audits->setParam($key, $value);
+        if (!empty($audits->getResource())) {
+            /**
+             * audits.payload is switched to default true
+             * in order to auto audit payload for all endpoints
+             */
+            $pattern = $route->getLabel('audits.payload', true);
+            if (!empty($pattern)) {
+                $audits->setPayload($responsePayload);
+            }
+
+            foreach ($events->getParams() as $key => $value) {
+                $audits->setParam($key, $value);
+            }
+            $audits->trigger();
         }
-        $audits->trigger();
-    }
-
-    if (!empty($deletes->getType())) {
-        $deletes->trigger();
-    }
-
-    if (!empty($database->getType())) {
-        $database->trigger();
-    }
-
-    if (
-        App::getEnv('_APP_USAGE_STATS', 'enabled') == 'enabled'
-        && $project->getId()
-        && $mode !== APP_MODE_ADMIN // TODO: add check to make sure user is admin
-        && !empty($route->getLabel('sdk.namespace', null))
-    ) { // Don't calculate console usage on admin mode
+
+        if (!empty($deletes->getType())) {
+            $deletes->trigger();
+        }
+
+        if (!empty($database->getType())) {
+            $database->trigger();
+        }
+
+        if (
+            App::getEnv('_APP_USAGE_STATS', 'enabled') == 'enabled'
+            && $project->getId()
+            && $mode !== APP_MODE_ADMIN // TODO: add check to make sure user is admin
+            && !empty($route->getLabel('sdk.namespace', null))
+        ) { // Don't calculate console usage on admin mode
+            $metric = $route->getLabel('usage.metric', '');
+            $usageParams = $route->getLabel('usage.params', []);
+
+            if (!empty($metric)) {
+                $usage->setParam($metric, 1);
+                foreach ($usageParams as $param => $value) {
+                    $parts = explode('.', $value);
+                    $namespace = $parts[0] ?? '';
+                    $key = $parts[1] ?? '';
+
+                    $params = [];
+                    switch ($namespace) {
+                        case 'response':
+                            $params = $responsePayload;
+                            break;
+                        case 'request':
+                            $params = $requestParams;
+                            break;
+                        case 'value':
+                            $usage->setParam($param, $key);
+                            break;
+                        default:
+                            $params = $responsePayload;
+                    }
+
+                    if (array_key_exists($key, $params)) {
+                        $usage->setParam($param, $params[$key]);
+                    }
+                }
+            }
```

Comment:
Would be great if we find a way to reuse this logic from audits logs. We can also tackle this later.

Reply:
Created a task to go back to this later.

Reply:
I've updated the code to match with audit labels, please check and verify.